### PR TITLE
Init CLI

### PR DESCRIPTION
**Pull request by YKo20010**
Added typer and commands for getting word or phrase associated to OLC.
Looks like this rn:
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.DS_Store
```
Comment (kx-chen) [inline .gitignore:1]: Looks like the original `.DS_STORE` is checked in, make sure to remove that too!

File: cli/main.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import sys
+import typer
+from functools import reduce
+
+sys.path.append('../')
+from web.mappings import map
```
Comment (kx-chen) [inline cli/main.py]:
```suggestion
from functools import reduce

import typer

from web.mappings import map
```

So apparently python's style guide suggests imports to be sorted in the following way: 
https://www.python.org/dev/peps/pep-0008/#imports

"Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.
You should put a blank line between each group of imports."

We don't need to worry too much about it though for this project.

File: cli/main.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import sys
+import typer
+from functools import reduce
+
+sys.path.append('../')
+from web.mappings import map 
+
+app = typer.Typer()
+
```
Comment (kx-chen) [inline cli/main.py:9]: Add an extra line. (stops linter from complaining)

File: cli/main.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+@app.command()
+def get_word(char_code: str):
+    word = map[char_code]
+    typer.echo(f"{word}")
+
```
Comment (kx-chen) [inline cli/main.py:15]: Add an extra line. (stops linter from complaining)

File: cli/main.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        char_arr = [code[i : i+2] for i in range(0, len(code), 2)]
+        # Convert coded array to word phrase.
+        phrase = reduce(lambda acc, x: acc + "." + map[x], char_arr, "")[1:]
+        typer.echo(f"{phrase}")
+    except:
+        error = typer.style("Not a valid OLC.", fg=typer.colors.RED)
+        typer.echo(error)
+
+if __name__ == "__main__":
+    app()
```
Comment (kx-chen) [inline cli/main.py]: Add an extra line. (stops linter from complaining)

File: cli/main.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        if len(code) < 8 != 0:
+            error = typer.style("OLC is too short.", fg=typer.colors.RED)
+            typer.echo(error)
+            return
+        # Split string into array of 2 char elements.
+        char_arr = [code[i : i+2] for i in range(0, len(code), 2)]
+        # Convert coded array to word phrase.
+        phrase = reduce(lambda acc, x: acc + "." + map[x], char_arr, "")[1:]
+        typer.echo(f"{phrase}")
+    except:
```
Comment (kx-chen) [inline cli/main.py]: Looks like this function throws a `KeyError`. Could we catch that just so it's more specific?

File: cli/main.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+@app.command()
+def get_phrase(code: str):
+    try:
+        if len(code) < 8 != 0:
+            error = typer.style("OLC is too short.", fg=typer.colors.RED)
+            typer.echo(error)
+            return
+        # Split string into array of 2 char elements.
+        char_arr = [code[i : i+2] for i in range(0, len(code), 2)]
```
Comment (kx-chen) [inline cli/main.py]:
```suggestion
        char_arr = [code[i: i+2] for i in range(0, len(code), 2)]
```

**Review by kx-chen — CHANGES_REQUESTED** [review]
Some nitpicky stuff (sorry!), but this is really well done! :)

File: cli/main.py (hunk 2 above)
Comment (kx-chen) [reply cli/main.py]: Also, it looks like with this change we can just run it directly using `python -m cli.main get-phrase 222426282`

File: .gitignore (hunk 1 above)
Comment (YKo20010) [reply .gitignore:1]: What do you mean by this? I don't see another .DS_Store in the gitignore.

File: cli/main.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import sys
```
Comment (chrisnguyn) [inline cli/main.py]: do we use sys? the linter is complaining it was imported but not used

File: .gitignore (hunk 1 above)
Comment (kx-chen) [reply .gitignore:1]: it's https://github.com/MLH-Fellowship/0.5.2-hackathon-geography/blob/0b3eb9711da3db318001ada996ab7c5af70ea65b/.DS_Store

**Review by kx-chen — APPROVED** [review]

**Pull request merged**